src/assisted_test_infra/test_infra/controllers/iscsi_target.py

```diff
@@ -0,0 +1,160 @@
+
+from assisted_test_infra.test_infra import utils
+
+from service_client import log
+
+
+class Iqn:
+
+    def __init__(self, unique_name: str, year_month: str = '2024-11', base_domain: str = 'iscsi.redhat.com'):
+        self.year_month = year_month
+        # base domain set in reverse order.
+        self.base_domain = '.'.join(base_domain.split('.')[::-1])
+        # node name + disk index may be used test-infra-cluster-4a25cd51-master-0-disk-0
+        self.unique_name = unique_name
+
+    def __repr__(self):
+        return f"iqn.{self.year_month}.{self.base_domain}:{self.unique_name}"
+
+    def __str__(self):
+        return f"iqn.{self.year_month}.{self.base_domain}:{self.unique_name}"
+
+
+class Receiver:
+
+    def __init__(self, fn_callback, **defaults_kwargs):
+        self.fn_callback = fn_callback
+        self.defaults_kwargs = defaults_kwargs
+
+    def execute(self, *args, **kwargs):
+        self.defaults_kwargs.update(kwargs)
+        log.info(f"iscsi_target exec: {str(args)} {str(kwargs)}")
+        self.fn_callback(*args, **kwargs)
+
+
+class IscsiTargetConfig:
+    def __init__(self, disk_name: str, disk_size_gb: int, iqn: Iqn, remote_iqn: Iqn, iso_disk_copy: str,
+                 servers: list[str]):
+        self._disk_name = disk_name
+        self._disk_size = disk_size_gb
+        self._iqn = iqn
+        self._data_file = iso_disk_copy
+        self._remote_iqn = remote_iqn
+        self._servers = servers
+
+    @property
+    def disk_name(self):
+        return self._disk_name
+
+    @property
+    def disk_size(self):
+        return self._disk_size
+
+    @property
+    def iqn(self):
+        return self._iqn
+
+    @property
+    def data_file(self):
+        return self._data_file
+
+    @property
+    def remote_iqn(self):
+        return self._remote_iqn
+
+    @property
+    def servers(self):
+        return self._servers
+
+
+class IscsiTargetController:
+
+    def __new__(cls):
+        # Make The controller singleton class because managed by hypervisor as a single service
+        if not hasattr(cls, 'instance'):
+            log.info('Creating singleton IscsiTargetController instance')
+            cls.instance = super(IscsiTargetController, cls).__new__(cls)
+        return cls.instance
+
+    def __init__(self):
+        self.builder = IscsiBuilder()
+        self._target_configs = []
+
+    def create_target_server(self, config: IscsiTargetConfig, clear_config=False) -> None:
```

Comment:
```suggestion
    def create_target(self, config: IscsiTargetConfig, clear_config=False) -> None:
```
target = server in iscsi vocabulary so let's use it once